homeassistant/components/fritz/common.py

```diff
@@ -285,3 +306,112 @@ def device_info(self):
             "model": self._fritzbox_tools.model,
             "sw_version": self._fritzbox_tools.sw_version,
         }
+
+
+class FritzBoxBaseSwitch:
+    """Fritz switch base class."""
+
+    def __init__(self, fritzbox_tools: FritzBoxTools, switch_info: SwitchInfo) -> None:
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
+        self._icon = switch_info["icon"]
+        self._type = switch_info["type"]
+        self._update: Callable = switch_info["callback_update"]
+        self._switch: Callable = switch_info["callback_switch"]
+
+        self._name = f"{self._friendly_name} {self._description}"
+        self._unique_id = (
+            f"{self.fritzbox_tools.unique_id}-{slugify(self._description)}"
+        )
+
+        self._attributes: dict[str, str] = {}
+        self._is_available = True
+        self._last_toggle_timestamp: datetime | None = None
+
+        self._attr_is_on = False
+
+    @property
+    def name(self):
+        """Return name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return name."""
+        return self._icon
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device information."""
+
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self.fritzbox_tools.mac)},
+            "name": self._friendly_name,
+            "manufacturer": "AVM",
+            "model": self.fritzbox_tools.model,
+            "sw_version": self.fritzbox_tools.sw_version,
+        }
+
+    @property
+    def available(self) -> bool:
+        """Return availability."""
+        return self._is_available
+
+    @property
+    def extra_state_attributes(self):
+        """Return device attributes."""
+        return self._attributes
+
+    async def async_update(self):
+        """Update data."""
+        if (
+            self._last_toggle_timestamp is not None
+            and (dt_util.utcnow() - self._last_toggle_timestamp).seconds < SWITCH_UPDATE
+        ):
+            _LOGGER.debug(
+                "Not updating switch state, because last toggle happened at %s, less than %s seconds ago",
+                self._last_toggle_timestamp.isoformat(sep=" ", timespec="seconds"),
+                SWITCH_UPDATE,
+            )
+            return
+
```

Comment:
Redundanz
```suggestion
```

Reply:
As per discord chat, this was due not to a API limit but to a old sync approach. Removed